runtime/vm/ContinuationHelpers.cpp

```diff
@@ -198,8 +223,18 @@ yieldContinuation(J9VMThread *currentThread)
 	 *
 	 * must be maintained for weakly ordered CPUs, to unsure that once the continuation is again available for GC scan (on potentially remote CPUs), all CPUs see up-to-date stack .
 	 */
-	Assert_VM_true((uintptr_t)currentThread == continuation->state);
-	continuation->state = J9_GC_CONTINUATION_STATE_CONCURRENT_SCAN_NONE;
+	Assert_VM_true(VM_VMHelpers::isContinuationMountedWithCarrierThread(continuation->state, currentThread));
+
+	j9object_t continuationObject = J9VMJAVALANGTHREAD_CONT(currentThread, currentThread->carrierThreadObject);
+	/* Notify GC of Continuation stack swap */
+	jboolean finished = J9VMJDKINTERNALVMCONTINUATION_FINISHED(currentThread, continuationObject);
+
+	if (!finished) {
+		VM_VMHelpers::resetContinuationStarted(continuation);
+		currentThread->javaVM->memoryManagerFunctions->postUnmountContinuation(currentThread, continuationObject);
+	} else {
+		continuation->state = VM_VMHelpers::setFinishedToContinuationState(continuation->state);
```

Comment:
this API is awkward; whole line should be just 

VM_VMHelpers::setContinuationFinished(continuation);

(if possible, inside the method it should assert 'started' is set)